Add Concepts: Supported Hosts (#241)
microsoft/vcpkg-docs: vcpkg/concepts/supported-hosts.md

---
title:  Supported Hosts
description: This article describes the platforms on which vcpkg is supported.
author: bion
ms.author: bion
ms.date: 01/16/2024
ms.topic: concept-article
---

# Supported Hosts

This document describes the architectures and operating systems supported by vcpkg. The examples
used in this document are current as of 2024-01-16.

## Dependencies

vcpkg also depends on a few system components to be installed before it can run.

To use vcpkg, a few components need to be installed on the system first.

* Windows: Visual Studio 2015 or later.
* macOS: XCode Console Tools, tar, git, zip, unzip, and curl. Most ports also need pkg-config.
We recommend installing these dependencies via Homebrew, such as via a terminal command
`brew install zip unzip curl pkgconfig`.
* Linux: A C++ compiler to use, tar, git, zip, unzip, and curl. Most ports also need pkg-config.
We recommend installing these dependencies with your system package manager.
  * apt-based platforms: `apt install git curl zip unzip pkgconfig`

>   * rpm-based platforms: `dnf install git curl zip unzip pkgconfig`

Reviewer: From the issues we know that macOS and Linux users don't get very far without `autoconf autoconf-archive automake make libtool`.

Reviewer: And maybe `python3`.

>   * rpm-based platforms: `dnf install git curl zip unzip pkgconfig`

Reviewer 2: ```suggestion
  * rpm-based platforms: `dnf install git curl zip unzip pkgconfig`

For Linux and macOS users we also recommend to install the following packages: `autoconf`, `autoconf-archive`, `automake`, `make` and `libtool`. While not required, these packages are used often by libraries available in vcpkg.
```

## Curated registry support levels

Support is broken into 4 distinct categories. They are:

* Full support, tested  

>   These configurations are those used to test vcpkg's curated registry on a regular basis. Problems in these
>   configurations are rapidly found and fixed by vcpkg's maintainers and community.

Reviewer 2: ```suggestion
  The configurations used to test vcpkg's curated registry on a regular basis. Problems in these configurations
  are rapidly found and fixed by vcpkg's maintainers and community..
```

* Full support, expected  

>   These configurations fixes are treated with the same level of priority as 'Full support, tested', but aren't used
>   directly to validate all changes to the curated registry. In general, these platforms are expected to be
>   substantially similar to one of the fully tested configurations.

Reviewer 2: ```suggestion
  Configurations treated with the same priority as 'Full support, tested' but which are not part of the regular tests.
  In general, these configurations are substantially similar to one of the fully tested configurations.
```

* Community support  

>   These configurations are not actively supported by vcpkg's maintainers, but have substantial community investment
>   where people other than the maintainers fix issues. In general, the vcpkg maintainers will merge changes to fix
>   problems for these platforms, but can't make statements that they should always work. The vcpkg maintainers are
>   interested in moving as many platforms out of community support into full support as possible.

Reviewer 2: ```suggestion
  Configurations that are not actively supported by vcpkg's maintainers but have substantial community investment.
  vcpkg maintainers accept changes to fix problems for these platforms but don't make guarantees that these 
  configurations work. The vcpkg team is interested in moving as many platforms out of community support
  into full support as possible.
```

* Unsupported  

>   These are platforms where things are known to not work for some reason. In general, PRs fixing only these platforms
>   aren't merged.

Reviewer 2: ```suggestion
  Configurations that are known no to work for some reason. In general, PRs fixing only these platforms
  are not accepted.
```

## Full support, tested

The fully supported, tested platforms are:

### Windows

* The latest Windows / Windows Server release. These are 11 and Server 2022 as of this writing.
* The latest Visual Studio update, Visual Studio 2022 17.8 as of this writing.

### macOS

macOS is intended to track the latest version of macOS and contemporary version of XCode Command Line tools. However,
updating macOS machines is a manual process, and macOS frequently changes things in ways that break vcpkg's testing. 
As of this writing, we are using:

* macOS 13.5.2
* XCode Command Line Tools 14.3.1

### Linux

* The latest LTS release of Ubuntu, currently 22.04.

> 

Reviewer 3: ```suggestion
### Android
* Linux 64-bit Android NDK version 25c
```
